### PR TITLE
upcoming: [M3-8370] – Add "Encryption" column to Linode Volumes table

### PR DESCRIPTION
**Pull request by dwiley-akamai**
## Description 📝
Add "Encryption" column to Linode Volumes table

## Target release date 🗓️
9/3/24

## Preview 📷
| Before  | After   |
| ------- | ------- |
| Screenshot 2024-08-16 at 2 53 04 PM | Screenshot 2024-08-16 at 2 53 40 PM |

### Prerequisites
Point at the dev environment with the `blockstorage-encryption` tag on your account

### Verification steps
Confirm the Linode Volumes table has the "Encryption" column when the feature flag is on. For any listed volumes, the encryption status should be accurate.

Unit test should pass:

```typescript
yarn test linodevolumes
```

## As an Author I have considered 🤔
- [X] 👀 Doing a self review
- [X] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [X] 🤏 Splitting feature into small PRs
- [X] ➕ Adding a changeset
- [X] 🧪 Providing/Improving test coverage
- [X] 🔐 Removing all sensitive information from the code and PR description
- [X] 🚩 Using a feature flag to protect the release
- [X] 👣 Providing comprehensive reproduction steps
- [X] 📑 Providing or updating our documentation
- [X] 🕛 Scheduling a pair reviewing session
- [X] 📱 Providing mobile support
- [X] ♿  Providing accessibility support

### CODE REVIEW

**Review by bnussman-akamai — COMMENTED** [review]
I'm seeing a broken table. Maybe a missing table cell?

Screenshot 2024-08-19 at 10 21 13 AM

**Comment by bnussman-akamai** [comment]
Any idea how to get past this error? I can attach unencrypted volumes, but when I try to attach an encrypted volume, I get this error. 

Screenshot 2024-08-19 at 10 29 06 AM

**Comment by dwiley-akamai** [comment]
> Any idea how to get past this error? I can attach unencrypted volumes, but when I try to attach an encrypted volume, I get this error.

I'm seeing the same. It's expected that a reboot would be required for linodes running on an unsupported qemu version, but provisioning and booting resources in alpha are in a weird state at the moment.

File: packages/manager/src/features/Linodes/LinodesDetail/LinodeStorage/LinodeVolumes.tsx [hunk 1]
```diff
@@ -124,6 +129,8 @@ export const LinodeVolumes = () => {
     return null;
   }
 
+  const numColumns = isBlockStorageEncryptionFeatureEnabled ? 6 : 5; // @TODO BSE: set colSpan for <TableRowEmpty /> to 6 once BSE is fully rolled out
```
Comment (mjac0bs) [inline packages/manager/src/features/Linodes/LinodesDetail/LinodeStorage/LinodeVolumes.tsx:132]: Can we fix the loading state table with the dynamic column number as well?

Feature flag off (no Encryption column):
Screenshot 2024-08-20 at 8 57 17 AM

Feature flag on (Encryption column):
Screenshot 2024-08-20 at 8 55 28 AM

**Review by mjac0bs — APPROVED** [review]
I was able to get past the reboot error with a new Linode created in Newark, that I then attached a single volume to. Other than the loading state table being short 1 column, the table is looking good.

Approving pending that's fixed and tests pass - I restarted the e2es because they failed before actually running.

❓ One question:

Are we missing the ability to encrypt a volume from the create drawer? I noticed this while testing the PR from the Linode Details page and glanced briefly through the list of remaining tickets in this epic - didn't see anything that looked like it covered this, just the clone drawer.

Screenshot 2024-08-20 at 9 07 14 AM

**Review by hana-linode — APPROVED** [review]
Confirmed encryption column in the Linode Volumes table and unit test passes. ✅ Approving pending the loading state table column is fixed

image

**Comment by dwiley-akamai** [comment]
> ❓ One question:
> 
> Are we missing the ability to encrypt a volume from the create drawer? I noticed this while testing the PR from the Linode Details page and glanced briefly through the list of remaining tickets in this epic - didn't see anything that looked like it covered this, just the clone drawer.

Thanks for raising this, I checked the wireframes and it's there, I think we may have overlooked it during the wireframes review. I'll create a ticket for it and get a PR up -- it should be pretty straightforward.

Latest commit fixes the loading state:

Screenshot 2024-08-20 at 1 28 09 PM

**Pull request merged**